Package Request: libopencv-dev

libopencv-dev please from 2.4.9 to 2.4.13, not 3.x branch.

I get

```
%] Building CXX object modules/core/CMakeFiles/opencv_core.dir/src/algorithm.cpp.o            clang-5.0: warning: argument unused during compilation: '-mfpu=neon' [-Wunused-command-line-argument]
In file included from /data/data/com.termux/files/home/opencv-2.4.13/modules/core/src/algorithm.cpp:43:                                            In file included from /data/data/com.termux/files/home/opencv-2.4.13/modules/core/src/precomp.hpp:48:                                              In file included from /data/data/com.termux/files/home/opencv-2.4.13/modules/core/include/opencv2/core/core.hpp:4921:                              /data/data/com.termux/files/home/opencv-2.4.13/modules/core/include/opencv2/core/operations.hpp:74:16: fatal error:                                      'ext/atomicity.h' file not found                 #include <ext/atomicity.h>                                ^~~~~~~~~~~~~~~~~                 1 error generated.                               make[2]: *** [modules/core/CMakeFiles/opencv_core.dir/build.make:63: modules/core/CMa
```

Sorry, for above dump, I'll format it later

> I have found it is easy to compile by myself, so this request is low priority:
> 
> ```
> apt install cmake pkg-config
> apt install libjpeg-turbo-dev libpng-dev
> wget -O opencv-2.4.13.zip http://sourceforge.net/projects/opencvlibrary/files/opencv-unix/2.4.13/opencv-2.4.13.zip/download
> unzip opencv-2.4.13
> cd opencv-2.4.13
> mkdir build
> cd build
> cmake -DCMAKE_BUILD_TYPE=RELEASE -DCMAKE_INSTALL_PREFIX=/data/data/com.termux/files/usr/local -DWITH_GTK=OFF -DWITH_1394=OFF -DBUILD_TESTS=OFF -DBUILD_PERF_TESTS=OFF -DWITH_JPEG=ON -DBUILD_JPEG=OFF ..
> make && make install
> ```
> 
> Now need to connect to camera somehow. I can use termux-api's `termux-camera-photo`, but it saves photo to a file which drains internal flash or MicroSD and overall this will be very slow. Need direct streaming from camera to OpenCV `VideoCapture::open()` somehow.

cmake -DCMAKE_BUILD_TYPE=RELEASE -DCMAKE_INSTALL_PREFIX=/data/data/com.termux/files/usr/local -DWITH_GTK=OFF -DWITH_1394=OFF -DBUILD_TESTS=OFF -DBUILD_PERF_TESTS=OFF -DWITH_JPEG=ON -DBUILD_JPEG=OFF ..                                 -- Detected processor: armv7l
-- Looking for ccache - not found
-- NEON is not supported by C++ compiler
-- FP16 is not supported by C++ compiler
-- Could NOT find OpenJPEG (minimal suitable version: 2.0, recommended version >= 2.3.1). OpenJPEG will be built from sources
-- OpenJPEG: VERSION = 2.3.1, BUILD = opencv-4.5.1-dev-openjp2-2.3.1
-- OpenJPEG libraries will be built from sources: libopenjp2 (version "2.3.1")
-- math lib 'libm' not found; floating point support disabled
CMake Error at cmake/android/OpenCVDetectAndroidSDK.cmake:49 (message):
  Android SDK: specify path to Android SDK via ANDROID_SDK_ROOT /
  ANDROID_HOME / ANDROID_SDK variables
Call Stack (most recent call first):
  cmake/android/OpenCVDetectAndroidSDK.cmake:172 (ocv_detect_android_sdk)
  CMakeLists.txt:723 (include)


-- Configuring incomplete, errors occurred!
See also "/data/data/com.termux/files/home/opencv/build/CMakeFiles/CMakeOutput.log".
See also "/data/data/com.termux/files/home/opencv/build/CMakeFiles/CMakeError.log".
$

cmake -DCMAKE_BUILD_TYPE=RELEASE -DCMAKE_INSTALL_PREFIX=/data/data/com.termux/files/usr/local -DWITH_GTK=OFF -DWITH_1394=OFF -DBUILD_TESTS=OFF -DBUILD_PERF_TESTS=OFF -DWITH_JPEG=ON -DBUILD_JPEG=OFF ..                                 -- Detected processor: armv7l
-- Looking for ccache - not found
-- NEON is not supported by C++ compiler
-- FP16 is not supported by C++ compiler
-- Could NOT find OpenJPEG (minimal suitable version: 2.0, recommended version >= 2.3.1). OpenJPEG will be built from sources
-- OpenJPEG: VERSION = 2.3.1, BUILD = opencv-4.5.1-dev-openjp2-2.3.1
-- OpenJPEG libraries will be built from sources: libopenjp2 (version "2.3.1")
-- math lib 'libm' not found; floating point support disabled
CMake Error at cmake/android/OpenCVDetectAndroidSDK.cmake:49 (message):
  Android SDK: specify path to Android SDK via ANDROID_SDK_ROOT /
  ANDROID_HOME / ANDROID_SDK variables
Call Stack (most recent call first):
  cmake/android/OpenCVDetectAndroidSDK.cmake:172 (ocv_detect_android_sdk)
  CMakeLists.txt:723 (include)


-- Configuring incomplete, errors occurred!
See also "/data/data/com.termux/files/home/opencv/build/CMakeFiles/CMakeOutput.log".
See also "/data/data/com.termux/files/home/opencv/build/CMakeFiles/CMakeError.log".
$

--- a/packages/opencv/build.sh
+++ b/packages/opencv/build.sh
@@ -14,6 +14,8 @@ TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
 "
 
 termux_step_pre_configure() {
+	termux_setup_protobuf
+
 	LDFLAGS+=" -llog"
 
 	find "$TERMUX_PKG_SRCDIR" -name CMakeLists.txt -o -name '*.cmake' | \

--- a/packages/opencv/build.sh
+++ b/packages/opencv/build.sh
@@ -1,0 +1,45 @@
+TERMUX_PKG_HOMEPAGE=https://opencv.org/
+TERMUX_PKG_DESCRIPTION="Open Source Computer Vision Library"
+TERMUX_PKG_LICENSE="Apache-2.0"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=4.5.5
+TERMUX_PKG_SRCURL=https://github.com/opencv/opencv/archive/${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=a1cfdcf6619387ca9e232687504da996aaa9f7b5689986b8331ec02cb61d28ad
+TERMUX_PKG_DEPENDS="libc++, libjpeg-turbo, libpng, libprotobuf, libtiff, libwebp, openjpeg, openjpeg-tools, zlib"
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
+-DANDROID_NO_TERMUX=OFF
+-DWITH_OPENEXR=OFF
+-DBUILD_PROTOBUF=OFF
+-DPROTOBUF_UPDATE_FILES=ON
+"
+
+termux_step_pre_configure() {
+	LDFLAGS+=" -llog"
+
+	find "$TERMUX_PKG_SRCDIR" -name CMakeLists.txt -o -name '*.cmake' | \
+		xargs -n 1 sed -i \
+		-e 's/\([^A-Za-z0-9_]ANDROID\)\([^A-Za-z0-9_]\)/\1_NO_TERMUX\2/g' \
+		-e 's/\([^A-Za-z0-9_]ANDROID\)$/\1_NO_TERMUX/g'
+
+	_NEED_DUMMY_LIBPTHREAD_A=
+	_LIBPTHREAD_A=$TERMUX_PREFIX/lib/libpthread.a
+	if [ ! -e $_LIBPTHREAD_A ]; then
+		_NEED_DUMMY_LIBPTHREAD_A=true
+		echo '!<arch>' > $_LIBPTHREAD_A
+	fi
+	_NEED_DUMMY_LIBRT_A=
+	_LIBRT_A=$TERMUX_PREFIX/lib/librt.a
+	if [ ! -e $_LIBRT_A ]; then
+		_NEED_DUMMY_LIBRT_A=true
+		echo '!<arch>' > $_LIBRT_A
+	fi
+}
+
+termux_step_post_make_install() {
+	if [ $_NEED_DUMMY_LIBPTHREAD_A ]; then
+		rm -f $_LIBPTHREAD_A
+	fi
+	if [ $_NEED_DUMMY_LIBRT_A ]; then
+		rm -f $_LIBRT_A
+	fi
+}

